### ISSUE TITLE
Bug Report: a11y - Techdocs code block copy to clipboard button doesn't have keyboard focus styling

### ISSUE THREAD

**Issue by logonoff** [backstage/backstage#29791]
### 📜 Issue Labels

- [x] Please familiarize yourself with the issue labels used in this project: [LABELS.md](https://github.com/backstage/backstage/blob/master/LABELS.md)

### 🔎 Search Terms

```plain
techdocs, keyboard, ctc, copy to clipboard
```

### 🗃️ Project Area

TechDocs

### 🔗 External Integration

N/A

### 📝 Description & Context

When navigating techdocs with a keyboard, the copy to clipboard button rendered inside code blocks does not receive styling when focused. 

### 👍 Expected Behavior

A focus ring or some other style appears when keyboard focus is given to the copy to clipboard button

### 🥾 Reproduction steps

1. Go to https://demo.backstage.io/docs/default/component/backstage/api/deprecations/
2. Use the keyboard to navigate the documentation
3. Observe the lack of focus styling in the copy to clipboard button

### Have you read the Code of Conduct?

- [x] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes, and I have enough information to get started
